lib/libvfio-user.c

```diff
@@ -487,7 +487,7 @@ handle_dma_map(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg,
     assert(msg != NULL);
     assert(region != NULL);
 
-    if (msg->in_size != sizeof(*region)) {
+    if (msg->in_size != sizeof(*region) || msg->in_size != region->argsz) {
```

Comment:
See my recent changes e.g. line :358, this should be:

```
if (msg->in_size < sizeof(*region) || in_info->argsz < sizeof(*region)) {
```

so client can specify a larger expected argsz

Reply:
ok

Reply:
Actually, I understand why `argsz` can be larger, but why `msg->in_size`? Won't we be ptentially masking errors if the client sends stuff that we ignore?